tests/std/tests/P2505R5_monadic_functions_for_std_expected/test.cpp

```diff
@@ -0,0 +1,377 @@
+            return expected<void, int>{};
+        } else {
+            return expected<Val, int>{Thingy{77}};
+        }
+    };
+    {
+        decltype(auto) result = forward<Expected>(engaged).or_else(to_expected_thingy);
+        static_assert(is_same_v<decltype(result), expected<Val, int>>);
+        assert(result);
+        if constexpr (!is_void_v<Val>) {
+            assert(result.value().x == 11);
+        }
+    }
+    {
+        decltype(auto) result = forward<Expected>(unengaged).or_else(to_expected_thingy);
+        static_assert(is_same_v<decltype(result), expected<Val, int>>);
+        assert(result);
+        if constexpr (!is_void_v<Val>) {
+            assert(result.value().x == 77);
+        }
+    }
+
+    engaged.transform([](auto...) { return ""; });
+}
+
+template <IsNothrowConstructible nothrowConstructible, IsNothrowConvertible nothrowConvertible>
+constexpr void test_error_or() {
+    constexpr bool construction_is_noexcept = IsYes(nothrowConstructible);
+    constexpr bool conversion_is_noexcept   = IsYes(nothrowConvertible);
+    constexpr bool should_be_noexcept       = construction_is_noexcept && conversion_is_noexcept;
+
+    struct payload_error_or {
+        constexpr payload_error_or() noexcept : _val(55) {}
+        constexpr payload_error_or(const int val) noexcept : _val(val) {}
+        constexpr payload_error_or(const payload_error_or& other) noexcept(construction_is_noexcept)
+            : _val(other._val + 2) {}
+        constexpr payload_error_or(payload_error_or&& other) noexcept(construction_is_noexcept)
+            : _val(other._val + 3) {}
+        constexpr payload_error_or(const convertible& val) noexcept(conversion_is_noexcept) : _val(val._val + 4) {}
+        constexpr payload_error_or(convertible&& val) noexcept(conversion_is_noexcept) : _val(val._val + 5) {}
+
+        [[nodiscard]] constexpr bool operator==(const payload_error_or&) const noexcept = default;
+
+        int _val;
+    };
+
+    { // with payload argument
+        using Expected = expected<int, payload_error_or>;
+
+        Expected with_error{unexpect, 42};
+        const Expected const_with_error{unexpect, 1337};
+        assert(with_error.error_or(payload_error_or{1}) == 42 + 2);
+        assert(const_with_error.error_or(payload_error_or{1}) == 1337 + 2);
+        static_assert(noexcept(with_error.error_or(payload_error_or{1})) == construction_is_noexcept);
+        static_assert(noexcept(const_with_error.error_or(payload_error_or{1})) == construction_is_noexcept);
+
+        assert(move(with_error).error_or(payload_error_or{1}) == 42 + 3);
+        assert(move(const_with_error).error_or(payload_error_or{1}) == 1337 + 2);
+        static_assert(noexcept(move(with_error).error_or(payload_error_or{1})) == construction_is_noexcept);
+        static_assert(noexcept(move(const_with_error).error_or(payload_error_or{1})) == construction_is_noexcept);
+
+        const payload_error_or input{2};

```

Comment:
In test code, I recommend "unique values for unique purposes". This makes it easy to see where values are coming from. This test has been excellent about following this principle, except here where `2` is identical to the `_val(other._val + 2)` in `payload_error_or`'s copy constructor. Occurs below.